src/instructlab/schema/v3/knowledge.json

```diff
@@ -0,0 +1,124 @@
+                    },
+                    "questions_and_answers": {
+                        "type": "array",
+                        "minItems": 3,
+                        "uniqueItems": true,
+                        "items": {
+                            "type": "object",
+                            "required": [
+                                "question",
+                                "answer"
+                            ],
+                            "properties": {
+                                "question": {
+                                    "description": "A question about the context which is used for synthetic data generation.",
+                                    "type": "string",
+                                    "minLength": 1
+                                },
+                                "answer": {
+                                    "description": "The desired response for the question.",
+                                    "type": "string",
+                                    "minLength": 1
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "document": {
+            "description": "The knowledge documents.",
+            "type": "object",
+            "required": [
+                "repo",
+                "commit",
+                "patterns"
+            ],
+            "unevaluatedProperties": false,
+            "properties": {
+                "repo": {
+                    "description": "The URL to a Git repository holding the knowledge documents.",
+                    "type": "string",
+                    "minLength": 1,
+                    "examples": [
+                        "https://github.com/instructlab/instructlab.git"
+                    ]
+                },
+                "commit": {
+                    "description": "The commit in the Git repository containing the knowledge documents.",
+                    "type": "string",
+                    "minLength": 1,
+                    "examples": [
+                        "951999afdc59c46d325493568193b40bd5439c9e"
+                    ]
+                },
+                "patterns": {
+                    "description": "An array of glob patterns of the knowledge documents in the Git repository.",
+                    "type": "array",
+                    "minItems": 1,
+                    "uniqueItems": true,
+                    "items": {
+                        "type": "string",
+                        "minLength": 1,
+                        "examples": [
+                            "*.md",
+                            "folder/*.md",
+                            "folder/knowledge_doc.md"
+                        ]
+                    }
+                }
+            }
+        },
+        "document_outline": {
+            "description": "An outline of the document.",
```

Comment:
If the document_outline is used anywhere in the prompt to the teacher model for SDG, we must be more precise in the description about how it is used and what people should put here. Otherwise people will treat it as a just some commentary field they just put something in to shut up the yaml validation.

See task_description in compositional skills.

Reply:
done, thanks!